java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JEnum.java

```diff
@@ -48,59 +47,45 @@ public String getType() {
+        return str.replaceAll("[\\s/]", "_");
     }
 
     @Override
     public GeneratorResult generateJava(CompilationUnit cu) {
-        EnumDeclaration en = cu.getEnumByName(this.type).orElse(null);
+        EnumDeclaration en = cu.addEnum(this.type);
 
-        if (en == null) {
-            en = cu.addEnum(this.type);
+        en.addField(JAVA_LANG_STRING, VALUE);
+        ConstructorDeclaration cd = en.addConstructor();
+        cd.addParameter(JAVA_LANG_STRING, VALUE);
+        cd.createBody();
 
-            boolean degraded = false;
-            for (String k : this.values) {
-                try {
-                    Integer.valueOf(k);
-                    degraded = true;
-                } catch (Exception e) {
-                    // Ignored
-                }
-            }
-
-            if (!degraded) {
-                for (String k : this.values) {
-                    en.addEnumConstant(sanitizeEnumEntry(sanitizeString(k)));
-                }
-            } else {
-                // TODO: test this properly eventually
-                en.addField(JAVA_LANG_STRING, VALUE);
-                ConstructorDeclaration cd = en.addConstructor();
-                cd.addParameter(JAVA_LANG_STRING, VALUE);
-                cd.createBody();
-
-                cd.setBody(
-                        new BlockStmt()
-                                .addStatement(
-                                        new AssignExpr(
-                                                new NameExpr("this." + VALUE),
-                                                new NameExpr(VALUE),
-                                                AssignExpr.Operator.ASSIGN)));
+        cd.setBody(
+                new BlockStmt()
+                        .addStatement(
+                                new AssignExpr(
+                                        new NameExpr("this." + VALUE),
+                                        new NameExpr(VALUE),
+                                        AssignExpr.Operator.ASSIGN)));
 
-                for (String k : this.values) {
-                    String constantName = sanitizeEnumEntry(sanitizeString(k));
-                    try {
-                        Integer.valueOf(k);
-                        constantName = "V_" + constantName;
-                    } catch (Exception e) {
-                        // Ignored
-                    }
-                    en.addEnumConstant(constantName + "(\"" + k + "\")");
-                }
+        for (String k : this.values) {
+            String constantName;
+            try {
+                // If the value can be parsed as an Integer
+                Integer.valueOf(k);
+                // Prepend
+                constantName = "V_" + sanitizeEnumEntry(sanitizeString(k));
+            } catch (Exception e) {
+                constantName = sanitizeEnumEntry(sanitizeString(k));
             }
+            EnumConstantDeclaration decl = new EnumConstantDeclaration();
```

Comment:
It might be interesting to make the constant name uppercase so that it complies with Java Naming convention.

Reply:
This is a very, very good use-case for introducing a `Config` class.
It's a decision that should belong to the user IMHO, let me do it 👍

Reply:
done, but this PR is growing 😊